bootloader/ubootenv/env.go

```diff
@@ -58,18 +56,38 @@ func writeUint32(u uint32) []byte {
 	return buf.Bytes()
 }
 
+const sizeOfUint32 = 4
+
+func calcHeaderSize(headerFlagByte bool) int {
+	if headerFlagByte {
+		// If uboot uses a header flag byte, header is 4 byte crc + flag byte
+		return sizeOfUint32 + 1
+	}
+	// otherwise, just a 4 byte crc
+	return sizeOfUint32
+}
+
+type CreateOptions struct {
+	HeaderFlagByte bool
+}
+
 // Create a new empty uboot env file with the given size
-func Create(fname string, size int) (*Env, error) {
+func Create(fname string, size int, opts *CreateOptions) (*Env, error) {
 	f, err := os.Create(fname)
 	if err != nil {
 		return nil, err
 	}
 	defer f.Close()
 
+	if opts == nil {
+		opts = &CreateOptions{HeaderFlagByte: true}
```

Comment:
this bit is not exercised

Reply:
Also if `HeaderFlagByte: true` is the default when not providing options, should this be documented somewhere?

Reply:
Added tests, and made the behavior explicit in the docstring